src/main/java/picocli/CommandLine.java

```diff
@@ -11861,11 +11855,11 @@ private boolean hasFullyMatchedSubgroup(boolean allRequired) {
             }
             private StringBuilder toString(StringBuilder result) {
                 int originalLength = result.length();
-                for (ArgSpec arg : originalStringValues.keySet()) {
-                    List<String> values = originalStringValues.get(arg);
+                for (Map.Entry<ArgSpec, List<String>> argEntry : originalStringValues.entrySet()) {
+                    List<String> values = argEntry.getValue();
```

Comment:
Same here: I prefer the previous iteration style since it is shorter and more readable.